### DISCUSSION TITLE
(Help) New to Malt trying to figure out outlines

### DISCUSSION THREAD

**Discussion by Jenssons** [bnpr/Malt#474] [Q&A]
ok so i know how to add outlines to the malt nodes but what im having a hard time figuring out is say my characters face is a separate model and material than the body so how would i make it so when you give them a outline it doesn't just put a line inbetween the face and neck. 

I would love to know how to add do line detection so they could be outlined like if they where one model. could you please have a image of the node setup please and thank you for reading

**Comment by dibli-goost** [comment]
Malt generates a unique Object ID for every object, which it uses to delineate between separate objects when generating line art.

You can replace the Object ID by using the "Pack ID" and "PRE PASS PIXEL SHADER Output" nodes.

These two spheres are separate objects with the same material. 

![image](https://user-images.githubusercontent.com/31767887/213894947-20c7c275-2e81-4360-897d-6b95ee7bc9c7.png)

We can replace the Object ID and Malt will treat them as if they are one object instead. 

![image](https://user-images.githubusercontent.com/31767887/213894923-1aa24091-b5de-478e-9c30-da1820054299.png)

**Reply by samytichadou** [reply]
Hi new malt user here, i'm unable to reproduce this setup, you can see in the screenshot this trick has no effect on the line processing

![image](https://github.com/user-attachments/assets/c064c28f-38b8-422e-8b8d-42c871edd342)

I may have missed something...

If not i'll report a bug, i am using latest blender/malt on windows 11 with a nvidia rtx3070ti laptop  (if this could be of any help :) )

**Reply by Jenssons** [reply]
Yeah I was having the same issue lately

**Reply by Jenssons** [reply]
@pragma37 any ideas why this has happen?

**Reply by pragma37** [reply]
Just remove the Depth and Normal Width.
And don't use (0,0,0,0) for the ID, that's the value used for the background.

Also, note that the ObjectID is also used for Self-Shadow detection, if you want the spheres to cast shadows onto each other you should use a CustomID for line detection instead (and set the first channel of `ID Boundary Width` to 0)

**Reply by Jenssons** [reply]
> Just remove the Depth and Normal Width. And don't use (0,0,0,0) for the ID, that's the value used for the background.
> 
> Also, note that the ObjectID is also used for Self-Shadow detection, if you want the spheres to cast shadows onto each other you should use a CustomID for line detection instead (and set the first channel of `ID Boundary Width` to 0)

im not sure what you mean here i still want depth lines, what this did before was basically treat all these different meshes like their one model so you could remove outlines on parts that shouldnt have them, like say this is on a head model and i didnt want the eyebrows to have a outline but the rest of the face to have them we would have just done this in the past and it worked

**Comment by Jenssons** [comment]
would it be possible to do something simalir with two different materials say if you have two objects with different texture images

**Reply by pragma37** [reply]
Yes, you just need to assign the same ID in both materials.
